extensions/binding/org.eclipse.smarthome.binding.dmx.test/src/test/java/org/eclipse/smarthome/binding/dmx/ArtnetBridgeHandlerTest.java

```diff
@@ -44,64 +41,57 @@
  *
  * @author Jan N. Klug - Initial contribution
  */
-public class ArtnetBridgeHandlerTest extends JavaOSGiTest {
+public class ArtnetBridgeHandlerTest extends JavaTest {
 
     private static final String TEST_ADDRESS = "localhost";
     private static final int TEST_UNIVERSE = 1;
-    private static final int TEST_CHANNEL = 100;
 
-    private ManagedThingProvider managedThingProvider;
-    private final VolatileStorageService volatileStorageService = new VolatileStorageService();
+    private final ThingUID BRIDGE_UID_ARTNET = new ThingUID(THING_TYPE_ARTNET_BRIDGE, "artnetbridge");
+    private final ChannelUID CHANNEL_UID_MUTE = new ChannelUID(BRIDGE_UID_ARTNET, CHANNEL_MUTE);
 
     Map<String, Object> bridgeProperties;
-    Map<String, Object> thingProperties;
 
     private Bridge bridge;
-    private Thing thing;
+    private ArtnetBridgeHandler bridgeHandler;
 
     @Before
     public void setUp() {
-        registerService(volatileStorageService);
-        managedThingProvider = getService(ThingProvider.class, ManagedThingProvider.class);
-
         bridgeProperties = new HashMap<>();
         bridgeProperties.put(CONFIG_ADDRESS, TEST_ADDRESS);
         bridgeProperties.put(CONFIG_UNIVERSE, TEST_UNIVERSE);
         bridge = BridgeBuilder.create(THING_TYPE_ARTNET_BRIDGE, "artnetbridge").withLabel("Artnet Bridge")
+                .withChannel(ChannelBuilder.create(CHANNEL_UID_MUTE, "Switch").withType(MUTE_CHANNEL_TYPEUID).build())
                 .withConfiguration(new Configuration(bridgeProperties)).build();
 
-        thingProperties = new HashMap<>();
-        thingProperties.put(CONFIG_DMX_ID, String.format("%d", TEST_CHANNEL));
-        thing = ThingBuilder.create(THING_TYPE_DIMMER, "testdimmer").withLabel("Dimmer Thing")
-                .withBridge(bridge.getUID()).withConfiguration(new Configuration(thingProperties)).build();
+        ThingHandlerCallback mockCallback = mock(ThingHandlerCallback.class);
+        doAnswer(answer -> {
+            ((Thing) answer.getArgument(0)).setStatusInfo(answer.getArgument(1));
+            return null;
+        }).when(mockCallback).statusUpdated(any(), any());
+
+        bridgeHandler = new ArtnetBridgeHandler(bridge) {
+            @Override
+            protected void validateConfigurationParameters(Map<String, Object> configurationParameters) {
```

Comment:
I would love to do so, however l. 318 in the BaseThingHandler.java (`ThingType thingType = thingTypeRegistry.getThingType(getThing().getThingTypeUID());`) throws a NPE. I assume this is due to the `thingTypeRegistry` is not available.

Reply:
That is correct. As I said before, for now it's not possible to full mock away all unnecessary stuff from the handlers :(

Apart from that, the code looks fine now.

I just wanted to double check with you if what you test is what you really wanted to test and not that you have removed half of you tests just because I mentioned that it can be done without inheriting from the `JavaOSGITest` class.